core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -2753,6 +2808,43 @@ private void resolveJoinColumns(QueryModel model) throws SqlException {
         }
     }
 
+    //Rewrite:
```

Comment:
What about expressions that include multiple columns like `sum(x+y)`? We could also rewrite those, but the below overflow concern would apply.

Reply:
Same as above .